test/SerialIOTest.cpp

```diff
@@ -907,10 +907,10 @@ TEST_CASE( "hdf5_dtype_test", "[serial][hdf5]" )
     REQUIRE(s.getAttribute("uchar").get< unsigned char >() == 'u');
     REQUIRE(s.getAttribute("int16").get< int16_t >() == 16);
     REQUIRE(s.getAttribute("int32").get< int32_t >() == 32);
-    REQUIRE(s.getAttribute("int64").get< int64_t >() == 64);
+    REQUIRE(getCast< int64_t >(s.getAttribute("int64")) == 64);
```

Comment:
As mentioned elsewhere, I would suggest we refactor `Variant` and `Attribute` to be closer / in one file.
With that, we could implement `getCast` alongside `get` directly in `openPMD::auxiliary::Variant`. This method will be useful for the user-facing code in many places.

Reply:
Agreed.
Is this `getCast` here for demonstration purposes? If not, why is it not used on the other explicit-width comparisons?

Reply:
Yes, demonstration and necessity.

> If not, why is it not used on the other explicit-width comparisons?

I don't get this, can you elaborate? Maybe I missed an already existing way to do this.

Reply:
Ok, more explicit this time.

Why is `getCast` used for `int64_t`, but not for `int32_t` or `int16_t`?
The only explaination I could come up with is the inconsistency between the `long` and `long long` aliasing, but this happens on a single platform where any stored HDF5 attribute should be read back in the `H5T_NATIVE_X`.

Reply:
Got ya, the reason is as follows: first of all, it should be everywhere to be maximally portable. That's why I propose to unify `auxiliary:Variant` with `Attribute` so that `Variant::get` can know all supported types already and we can implement the external `getCast` *inside* the current `Variant::get` for maximum user comfort :)

I am only changing the `int64_t` in this PR, since on OSX 64bit, which has like Linux 64bit `long` and `long long` as `int64_t`, the `int64_t` alias is not on `long long` but on `long`, thus returning those as `long`.

Reply:
That's the same reason why #184 fails, so we can fix it the same way for `long double` on MSVC :)

Reply:
I agree with your proposal as that will solve the problem in the general case.

I just don't grasp all the details of the problem, still. :/
So this one cast is (with our current platform exploration) mandatory only for `int64_t` as that is stored in the `Attibute` as `long long`, but the `Variant::get< int64_t >()` is equivalent to a `Variant::get< long >()`.

... Right?

Reply:
Yes, it is mandatory since a `long long` will round-trip as `long` on OSX during write->read.
We always (consistently) choose the "smallest" generic type that fits, that means if a `long long` is also a `long` a written `long long` will be returned as `long`. To hide this nastiness and it's implications for `std::variant` (not an alias == bad access) from the user, we use a static cast for explicitly possible types.

We can decide if we allow precision loss in `get<>`. E.g., only exact same implementations vs. all possible `static_cast`s. I think even the latter is fine, a user can check the type at runtime.

Reply:
There is already a similar casting and reporting mechanism in place for floating point attributes that the openPMD standard does not fix in bid-width. This might become obsolete in the process of implementing this extended `Variant::get< T >()`.
See [here](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/src/backend/Attributable.cpp#L282-L290).